proposals/unsafe-evolution.md

```diff
@@ -0,0 +1,138 @@
+# Unsafe Evolution
+
+Champion issue: https://github.com/dotnet/csharplang/issues/9704
+
+## Summary
+
+We update the definition of `unsafe` in C# from refering to locations where pointer types are used, to be locations where memory unmanaged by the runtime is dereferenced. These locations
+are where memory unsafety occurs, and are responsible for the bulk of CVEs categorized as memory safety issues.
+
+```cs
+void M()
+{
+    int i = 1;
+    int* ptr = &i; // Not unsafe
+    unsafe
+    {
+        Console.WriteLine(*ptr); // Dereference of memory not managed by the runtime. This is unsafe.
+        ref int intRef = Unsafe.AsRef(ptr); // Conversion of memory not managed by the runtime to a `ref`. This is unsafe.
+    }
+}
+```
+
+## Motivation
+
+Background for this feature can also be found in https://github.com/dotnet/designs/pull/330, which tracks the broader ecosystem changes that will be needed as part of this proposal. For C#
+specifically, we want to make sure that memory unsafety is properly tracked by the language; today, it can be difficult to look at a program hollistically and understand all locations where
```

Comment:
```suggestion
specifically, we want to make sure that memory unsafety is properly tracked by the language; today, it can be difficult to look at a program holistically and understand all locations where
```